### ISSUE TITLE
`@stylistic/declaration-block-semicolon-newline-before` takes the semicolon into an inline comment under `never-multi-line`

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#207]
Found while fixing [#203](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/203), and unrelated to it: the `always` options are not affected, since the break they write is what closes such a comment anyway. This is the shape [#117](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/117) had for the `declaration-block-semicolon-space-before` sibling.

### What minimal example or steps are needed to reproduce the bug?

```scss
a {
  color: red // keep me
  ;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "customSyntax": "postcss-scss",
  "rules": {
    "@stylistic/declaration-block-semicolon-newline-before": "never-multi-line"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.scss" --config myconfig.json --fix
```

### What did you expect to happen?

The problem is reported and the code left as it is: an inline comment ends only with a line break, so there is nowhere to put the semicolon that the comment would not swallow.

### What actually happened?

The semicolon is pulled up onto the comment's line and commented out, the declaration ends nowhere, and nothing is reported:

```scss
a {
  color: red // keep me;
}
```

The same happens under `postcss-less`, and where the comment stands behind `!important`: `color: red !important // keep me` followed by `;` becomes `color: red !important // keep me;`.

### What is the cause?

The `never-multi-line` branch trims the trailing whitespace of the value, or of the flag's raw, with `\s*$`, which counts the line break closing an inline comment as ordinary whitespace. The sibling declines the fix where the text it writes ends with such a comment, with `endsWithInlineComment`, and lets the warning stand; this rule wants the same guard in that branch.
